retrieval/scrape.go

```diff
@@ -807,8 +879,11 @@ func (sl *scrapeLoop) reportStale(start time.Time) error {
 }
 
 func (sl *scrapeLoop) addReportSample(app storage.Appender, s string, t int64, v float64) error {
-	ref, ok := sl.refCache[s]
+	// Suffix s with the invalid \xff unicode rune to avoid collisions
+	// with scraped metrics.
+	s2 := s + "\xff"
```

Comment:
It will usually collide, however it's possible the user is ingesting an "up" metric from the scrape that will not end up with the same labels as the target and thus not collide.